packages/vite/src/node/ssr/ssrTransform.ts

```diff
@@ -37,17 +38,31 @@ export async function ssrTransform(
     locations: true
   }) as any
 
-  let uid = 0
   const deps = new Set<string>()
   const idToImportMap = new Map<string, string>()
   const declaredConst = new Set<string>()
+  const usedNames = new Set<string>()
+
+  function findFreeName(name: string) {
+    let suffix = 0
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
+      const usedName = ++suffix > 1 ? `${name}_${suffix}` : name
+      if (!usedNames.has(usedName)) {
+        usedNames.add(usedName)
+        return usedName
+      }
+    }
+  }
 
   function defineImport(node: Node, source: string) {
     deps.add(source)
-    const importId = `__vite_ssr_import_${uid++}__`
+    const importId = findFreeName(
+      `import_` + path.basename(source, path.extname(source)).replace(/\W/g, '')
```

Comment:
Should we replace with `_` instead of space so it looks better with subdependencies? (maybe a good idea to add a test case for them?)
```js
import {error} from 'dependency/sub'
```

Reply:
Ah... in this case it will only use `sub` because of using basename, no? There may be more collisions than when using the full path in that case but it is true that we should probably have a way to limit by length if we do that

Reply:
It's hard to know when to stop, so I think basename is enough. This is especially true when subpaths are used:
```ts
import Button from '@foo/bar/components/Button'
// ..becomes which one?
const import_Button = require(...)
const import_components_Button = require(...)
const import_bar_components_Button = require(...)
const import_foo__bar_components_Button = require(...)
```

There won't be collisions in the case of duplicate basenames, because `findFreeName` checks for collisions.